pkg/operation/botanist/controlplane.go

```diff
@@ -857,6 +850,21 @@ func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
 			},
 		}
 
+		seedVersionGE112, err := version.CompareVersions(b.K8sSeedClient.Version(), ">=", "1.12")
```

Comment:
```suggestion
		seedVersionGE112, err := version.CompareVersions(b.K8sSeedClient.Version(), ">=", "1.12-0")
```
IIRC, this is necessary to also support versions used by other providers e.g., `v1.18.14-gke.1200`.

Reply:
With https://github.com/gardener/gardener/pull/3444 this will be handled transparently so there is no need to have the `-0` suffix.

Reply:
Thanks for checking the details :)